src/tools/pkcs11-tool.c

```diff
@@ -7965,9 +7979,13 @@ static struct mech_info	p11_mechanisms[] = {
       { CKM_SHA512,		"SHA512", NULL, MF_UNKNOWN },
       { CKM_SHA512_HMAC,	"SHA512-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_224,		"SHA3-224", NULL, MF_UNKNOWN },
+      { CKM_SHA3_224_HMAC,	"SHA3-224-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_256,		"SHA3-256", NULL, MF_UNKNOWN },
+      { CKM_SHA3_256_HMAC,	"SHA3-256-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_384,		"SHA3-384", NULL, MF_UNKNOWN },
+      { CKM_SHA3_384_HMAC,	"SHA3-384-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_SHA3_512,		"SHA3-512", NULL, MF_UNKNOWN },
+      { CKM_SHA3_512_HMAC,	"SHA3-512-HMAC", NULL, MF_GENERIC_HMAC_FLAGS },
       { CKM_RIPEMD128,		"RIPEMD128", NULL, MF_UNKNOWN },
       { CKM_RIPEMD128_HMAC,	"RIPEMD128-HMAC", NULL, MF_UNKNOWN },
       { CKM_RIPEMD128_HMAC_GENERAL,"RIPEMD128-HMAC-GENERAL", NULL, MF_UNKNOWN },
```

Comment:
I can't test it but we also need MF_GENERIC_HMAC_FLAGS here

Reply:
please, add them also to any other mechanism that will require them even if you can not test it. I do not think the MD5 nor RIPEMD128 will be used somewhere in the future for anything useful, but if somebody will needed, they should not be blocked by the tool

Reply:
Please, have a look at my fourth commit.

Reply:
looks good.